pkg/agent/kube.go

```diff
@@ -23,11 +24,34 @@ type ClusterKubeAPIClient struct {
 	configPath string
 }
 
+// BuildConfigFromFlags is a helper function that builds configs from a master
+// url or a kubeconfig filepath. These are passed in as command line flags for cluster
+// components. Warnings should reflect this usage. If neither masterUrl or kubeconfigPath
+// are passed in we fallback to inClusterConfig. If inClusterConfig fails, we fallback
+// to the default config.
+//
+// From https://github.com/kubernetes/client-go/blob/ab86e03da476bd58948f318b4f73d67d0801fcbd/tools/clientcmd/client_config.go#L657
+// but modifies the "Neither --kubeconfig nor --master was specified.  Using the
+// inClusterConfig.  This might not work." warning.
+func BuildConfigFromFlags(masterUrl, kubeconfigPath string) (*rest.Config, error) {
+	if kubeconfigPath == "" && masterUrl == "" {
```

Comment:
What about using something like https://github.com/openshift/installer/blob/36bbe11d2c1b1aa0891627265e69c59d82f4ba78/pkg/asset/agent/joiner/clusterinfo.go#L130-L134?

Reply:
Yes, this is better.